plugins/dbviewer/api/api.js

```diff
@@ -1,18 +1,31 @@
 var common = require('../../../api/utils/common.js'),
+    log = common.log('dbviewer:api'),
     async = require('async'),
     plugins = require('../../pluginManager.js'),
     countlyFs = require('../../../api/utils/countlyFs.js'),
     _ = require('underscore'),
     taskManager = require('../../../api/utils/taskmanager.js'),
-    {dbUserHasAccessToCollection, dbLoadEventsData} = require('../../../api/utils/rights.js'),
+    { dbUserHasAccessToCollection, dbLoadEventsData } = require('../../../api/utils/rights.js'),
+    { validateGlobalAdmin } = require('../../../api/utils/rights.js'),
     exported = {};
 
+var spawn = require('child_process').spawn,
+    child;
 (function() {
     plugins.register("/o/db", function(ob) {
-        var dbs = {countly: common.db, countly_drill: common.drillDb, countly_out: common.outDb, countly_fs: countlyFs.gridfs.getHandler()};
+        var dbs = { countly: common.db, countly_drill: common.drillDb, countly_out: common.outDb, countly_fs: countlyFs.gridfs.getHandler() };
         var params = ob.params;
+        var paths = ob.paths;
         var dbNameOnParam = params.qstring.dbs || params.qstring.db;
-
+        var validateUserForDataReadAPI = ob.validateUserForDataReadAPI;
+        if (paths[3]) {
+            switch (paths[3]) {
+            case 'mongotop': return validateUserForDataReadAPI(params, fetchMongoTop);
```

Comment:
validateGlobalAdmin should be added here.
Currently you first validate user for Data read API and then in the fetchMongoTop function validate user for global admin.
You only need to validate the user for global admin.